src/main/java/net/minestom/server/MinecraftServer.java

```diff
@@ -742,7 +745,10 @@ public static void setShouldProcessNettyErrors(boolean processNettyErrors) {
      * @param port                 the server port
      * @param responseDataConsumer the response data consumer, can be null
      * @throws IllegalStateException if called before {@link #init()} or if the server is already running
+     *
+     * @deprecated use {@link #start(String, int)} and listen to the {@link net.minestom.server.event.server.StatusRequestEvent} event instead of ResponseDataConsumer
      */
+    @Deprecated
     public void start(@NotNull String address, int port, @Nullable ResponseDataConsumer responseDataConsumer) {
```

Comment:
This method could overload the other one? Just set the consumer and then call the other init method.